src/Microsoft.AspNet.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -33,20 +33,30 @@ public static class TagHelperOutputExtensions
         {
             if (!tagHelperOutput.Attributes.ContainsName(attributeName))
             {
-                IEnumerable<IReadOnlyTagHelperAttribute> entries;
+                var copiedAttribute = false;
 
-                // We look for the original attribute so we can restore the exact attribute name the user typed.
-                // Approach also ignores changes made to tagHelperOutput[attributeName].
-                if (!context.AllAttributes.TryGetAttributes(attributeName, out entries))
+                // We iterate context.AllAttributes backwards since we prioritize TagHelperOutput values occurring
+                // before the current context.AllAttribtes[i].
+                for (var i = context.AllAttributes.Count - 1; i >= 0; i--)
                 {
-                    throw new ArgumentException(
-                        Resources.FormatTagHelperOutput_AttributeDoesNotExist(attributeName, nameof(TagHelperContext)),
-                        nameof(attributeName));
+                    // We look for the original attribute so we can restore the exact attribute name the user typed in
+                    // approximately the same position where the user wrote it in the Razor source.
+                    // Approach also ignores changes made to tagHelperOutput[attributeName].
```

Comment:
What `changes made to tagHelperOutput[attributeName]`? Have already confirmed this `attributeName` does not appear in the output collection.